w-near-141/src/fungible_token_core.rs

```diff
@@ -171,7 +176,10 @@ impl FungibleTokenResolver for Contract {
                     amount
                 }
             }
-            PromiseResult::Failed => amount,
+            PromiseResult::Failed => {
+                Promise::new(sender_id.to_string()).transfer(initial_storage_deposit.into());
```

Comment:
This can be abused, since the receiver can return a promise that fails, which doesn't generate a refund.

So this fungible contract core implementation can't be sure that there was refund.

It's a limitation of NEAR protocol cross-contracts right now.

Reply:
Yup agreed, reset to `8da15fa`